omod-common/src/main/java/org/openmrs/module/webservices/rest/web/RestUtil.java

```diff
@@ -936,4 +940,8 @@ public static Class<?> getSupportedClass(Resource resource) {
 			        .supportedClass();
 		}
 	}
+
+	public static boolean isUuid(String str) {
+		return StringUtils.isNotBlank(str) && UUID_REGEX.matcher(str).matches();
```

Comment:
I couldn't find any existing standard util methods for this (besides doing a UUID.toString() in a try/catch block, which was ugly) so I just grabbed a regex I found online.

Reply:
I've definitely seen other implementations of this in OpenMRS code, can't remember where off-hand...

Reply:
There are a few lying around, but they often run up against the issue that we've been allowing invalid UUIDs for years (e.g., every single "uuid" in CIEL is, in fact, not a "valid UUID"). Usually these implementations hang around until someone complains and we basically end up with a check that says "is this string 36 characters long"? [Here is one version](https://github.com/openmrs/openmrs-module-emrapi/blob/ae67609c476906e6fc5fdf57c4a97cd6f8734e63/api/src/main/java/org/openmrs/module/emrapi/utils/GeneralUtils.java#L163) and a [slight variant](https://github.com/openmrs/openmrs-module-webservices.rest/blob/4f3e8ecd078bfde7870dada85f1b2521e9e85048/omod-1.9/src/main/java/org/openmrs/module/webservices/rest/web/v1_0/controller/openmrs1_9/ConceptReferenceController1_9.java#L98).